File: 70. Climbing Stairs.py
class Solution:
    def climbStairs(self, n: 'int') -> 'int':
        dp=[]
        dp.append(1)
        dp.append(2)
        if n==1 or n==2:
            return n
        for i in range(2,n):
            dp.append(dp[i-1] + dp[i-2])
        return dp[-1]

File: 70. Climbing Stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        std::vector<int> dp(n+3, 0);
        dp[1] = 1;
        dp[2] = 2;
        for(int i=3; i <=n; i++){
            dp[i] = dp[i-2] + dp[i-1];
        }
        return dp[n];
    }
};
